corehq/form_processor/tests/utils.py

```diff
@@ -135,20 +103,34 @@ def delete_all_sql_forms(domain=None, user_id=None):
     @unit_testing_only
     def delete_all_sync_logs(cls):
         logger.debug("Deleting all synclogs")
-        cls._delete_all(SyncLog.get_db(), 'phone/sync_logs_by_user')
+        cls._delete_all(SyncLog.get_db(), ["SyncLog"])
 
     @staticmethod
     @unit_testing_only
-    def _delete_all(db, viewname, **view_kwargs):
-        deleted = set()
-        for row in db.view(viewname, reduce=False, **view_kwargs):
-            doc_id = row['id']
-            if doc_id not in deleted:
-                try:
-                    safe_delete(db, doc_id)
-                    deleted.add(doc_id)
-                except ResourceNotFound:
-                    pass
+    def _delete_all(db, doc_types, domain=None):
+        for doc_type in doc_types:
+            if domain:
+                view = 'by_domain_doc_type_date/view'
+                view_kwargs = {
+                    'startkey': [domain, doc_type],
+                    'endkey': [domain, doc_type, {}],
+                }
+            else:
+                view = 'all_docs/by_doc_type'
+                view_kwargs = {
+                    'startkey': [doc_type],
+                    'endkey': [doc_type, {}],
+                }
+
+            deleted = set()
+            for row in db.view(view, reduce=False, **view_kwargs):
+                doc_id = row['id']
+                if doc_id not in deleted:
+                    try:
+                        safe_delete(db, doc_id)
+                        deleted.add(doc_id)
+                    except ResourceNotFound:
+                        pass
```

Comment:
Previously this wasn't deleting 'deleted' docs because it was using views that excluded them.